augur/export_v2.py

```diff
@@ -46,51 +46,67 @@ def convert_tree_to_json_structure(node, metadata, div=0, strains=None):
     return (node_struct, strains)
 
 
-def process_colorings(json: list, color_mapping, nodes=None, node_metadata: dict=None) -> dict:
-    if not json:
-        print("WARNING: no colorings were defined")
-        return
-
-    data = {k: {'type': 'categorical'} for k in json}
-    #Always add in genotype and date to colour by
-    data['gt'] = {'title': 'Genotype', 'type': 'ordinal'}
-    #figure out how to see if num_date is there? Is it possible to not have?
-    data['num_date']  = {'title': 'Sampling date', 'type': 'continuous'}
-
-    if 'clade_membership' in data and 'title' not in data['clade_membership']:
-        data['clade_membership']['title'] = 'Clade'
-
-    for trait, options in data.items():
-        if "type" not in options:
-            raise Exception("coloring {} missing type...".format(trait))
-
-        if "title" not in options:
-            options["title"] = trait
-
-        if nodes:
-            values_in_tree = {node[trait] for node in nodes.values() if trait in node}
-        else:
-            values_in_tree = set()
-            for name, values in node_metadata.items():
-                if trait in values['traits']:
-                    values_in_tree.add(values['traits'][trait]['value'])
-
-        if trait.lower() in color_mapping:
-            # remember that the color maps (from the TSV) are in lower case, but
-            # this is not how they should be exported
-            case_map = {str(val).lower(): val for val in values_in_tree}
-            options["scale"] = {case_map[m[0]]: m[1] for m in color_mapping[trait.lower()] if m[0] in case_map}
-
-        else:
-            # TODO detect other types (ordinal, boolean)
-            if options['type'] == 'categorical' and options['title'] != 'authors':
-                #if its int or float and no colours supplied - set continuous.
-                # TODO this should maybe be expanded to long and scientific notation (?)
-                if all([ isinstance(n, float) if isinstance(n, float) else isinstance(n, int) for n in values_in_tree ]):
+def get_colorings(config, traits, provided_colors, node_metadata, mutations_present):
+    def _get_values(key):
+        values_in_tree = set()
+        for node_properties in node_metadata.values():
+            # beware of special cases!
+            if key in ["num_date", "author"]:
+                values_in_tree.add(node_properties.get(key, {}).get("value"))
+            elif key in node_properties['traits']:
+                values_in_tree.add(node_properties['traits'][key]['value'])
+        return values_in_tree
+
+    def _guess_type(key, values):
+        # TODO detect other types (ordinal, boolean)
+        if all([ isinstance(n, float) if isinstance(n, float) else isinstance(n, int) for n in values ]):
+            return "continuous"
+        return "categorical"
+
+    def _guess_title(key):
```

Comment:
Probably good to incorporate here my guessing from old v1 options `menuItem` & `legendTitle`. Here or below should also print a warning and remove these two so they don't get written to JSON.

Reply:
Done in merge 😄